tensorflow_addons/optimizers/tests/gradient_accumulator_test.py

```diff
@@ -225,6 +271,7 @@ def _generator_fn():
         inputs = layers.Input(shape=(None,), dtype=tf.int32)
         x = inputs
         x = layers.Embedding(input_dim=vocab_size, output_dim=8)(x)
+        x = layers.LSTM(4)(x)
```

Comment:
This layer seems to make the training crash. I am able to use `GradientAccumulator` in my ASR training but for some reason this test crashes:

### Log

```none
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

op_name = b'CudnnRNN', num_outputs = 4
inputs = [<tf.Tensor: shape=(9, 0, 8), dtype=float32, numpy=array([], shape=(9, 0, 8), dtype=float32)>, <tf.Tensor: shape=(1, 0...00,  0.00000000e+00,
        0.00000000e+00,  0.00000000e+00,  0.00000000e+00,  0.00000000e+00],
      dtype=float32)>]
attrs = ('T', 1, 'rnn_mode', b'lstm', 'input_mode', b'linear_input', ...)
ctx = <tensorflow.python.eager.context.Context object at 0x7f18872c36d0>
name = None

    def quick_execute(op_name, num_outputs, inputs, attrs, ctx, name=None):
      """Execute a TensorFlow operation.
    
      Args:
        op_name: Name of the TensorFlow operation (see REGISTER_OP in C++ code) to
          execute.
        num_outputs: The number of outputs of the operation to fetch.
                     (Explicitly provided instead of being inferred for performance
                     reasons).
        inputs: A list of inputs to the operation. Each entry should be a Tensor, or
          a value which can be passed to the Tensor constructor to create one.
        attrs: A tuple with alternating string attr names and attr values for this
          operation.
        ctx: The value of context.context().
        name: Customized name for the operation.
    
      Returns:
        List of output Tensor objects. The list is empty if there are no outputs
    
      Raises:
        An exception on error.
      """
      device_name = ctx.device_name
      # pylint: disable=protected-access
      try:
        ctx.ensure_initialized()
>       tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
                                            inputs, attrs, num_outputs)
E                                           tensorflow.python.framework.errors_impl.UnknownError: CUDNN_STATUS_BAD_PARAM
E                                           in tensorflow/stream_executor/cuda/cuda_dnn.cc(1514): 'cudnnSetTensorNdDescriptor( tensor_desc.get(), data_type, sizeof(dims) / sizeof(dims[0]), dims, strides)' [Op:CudnnRNN]

/home/sfalk/miniconda3/envs/addons/lib/python3.8/site-packages/tensorflow/python/eager/execute.py:59: UnknownError
```

Reply:
This is resolved. I was using the wrong strategy here.

Reply:
Yeah, but it runs on single gpu now.

Reply:
Yeah I know. You said we cannot test on multiple GPUs anyway, right? Only virtual ones. The accumulator works in my ASR training. :)